### PR TITLE
Update BabylonNative submodule and Babylon.js package version

### PR DESCRIPTION
**Pull request by bghgary**
**Describe the change**

Update Babylon Native submodule and Babylon.js dependency versions. There was a PROTOCOL_VERSION update and a bgfx update.

**Testing**

- [x] Android
- [x] iOS

### CODE REVIEW

**Review by CoPrez — APPROVED** [review]

**Review by CedricGuillemet — APPROVED** [review]
LGTM!

File: Apps/PackageTest/0.63.1/package.json [hunk 1]
```diff
@@ -10,8 +10,8 @@
     "lint": "eslint . --ext .js,.jsx,.ts,.tsx"
   },
   "dependencies": {
-    "@babylonjs/core": "5.6.1",
-    "@babylonjs/react-native": "^0.4.0-alpha.47",
+    "@babylonjs/core": "5.35.1",
+    "@babylonjs/react-native": "^1.4.1",
```
[the remark this answers is not in the document]
Comment (ryantrem) [reply Apps/PackageTest/0.63.1/package.json:14]: I'm guessing this is broken since it doesn't even reference any of the BRN binary packages. But yea removing it or fixing it in a separate PR makes sense.

File: Apps/Playground/0.65/ios/Playground.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -485,6 +484,7 @@
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
 				CLANG_ENABLE_MODULES = YES;
 				CURRENT_PROJECT_VERSION = 1;
+				DEVELOPMENT_TEAM = UBF8T346G9;
```
Comment (ryantrem) [inline Apps/Playground/0.65/ios/Playground.xcodeproj/project.pbxproj:487]: I don't think you want to check in these `DEVELOPMENT_TEAM` changes.

File: Apps/Playground/0.65/npm_run_start_loop.cmd [hunk 3]
```diff
@@ -0,0 +1,6 @@
+:Loop
```
Comment (ryantrem) [inline Apps/Playground/0.65/npm_run_start_loop.cmd:1]: What is this? If we want some tooling like this, we should put it in a shared node script somewhere.

File: Modules/@babylonjs/react-native-iosandroid/android/build.gradle [hunk 4]
```diff
@@ -94,7 +94,6 @@ android {
     }
     externalNativeBuild {
         cmake {
-            version '3.19.6'
```
Comment (ryantrem) [inline Modules/@babylonjs/react-native-iosandroid/android/build.gradle:97]: Why this change?

File: Apps/Playground/0.65/npm_run_start_loop.cmd (hunk 3 above)
Comment (bghgary) [reply Apps/Playground/0.65/npm_run_start_loop.cmd:1]: Oops, that's not supposed to be in the PR. :)

File: Apps/Playground/0.65/ios/Playground.xcodeproj/project.pbxproj (hunk 2 above)
Comment (bghgary) [reply Apps/Playground/0.65/ios/Playground.xcodeproj/project.pbxproj:487]: Well, that's annoying. We have to undo the dev team before merging? I've never had to do that. I wonder what's different?

File: Modules/@babylonjs/react-native-iosandroid/android/build.gradle (hunk 4 above)
Comment (bghgary) [reply Modules/@babylonjs/react-native-iosandroid/android/build.gradle:97]: Ugh, sorry, this isn't supposed to be there either. I run with a newer version of cmake.

File: Apps/Playground/0.65/ios/Playground.xcodeproj/project.pbxproj (hunk 2 above)
Comment (bghgary) [reply Apps/Playground/0.65/ios/Playground.xcodeproj/project.pbxproj]: Weird, I don't know what happened, but it didn't add them this time. I was having trouble with setting the dev team, so maybe I touched something that caused xcode to write it out. Anyways, it's removed now.

File: Apps/PackageTest/0.63.1/package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (bghgary) [reply Apps/PackageTest/0.63.1/package.json:14]: We can do it in a separate PR.
